tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2785,13 +2785,17 @@ AddBuiltInTriangleIntersectionAttributes(ASTContext &context,
 //
 // Subobjects
 
-static CXXRecordDecl *StartSubobjectDecl(ASTContext &context,
-                                         const char *name) {
+static CXXRecordDecl *
+StartSubobjectDecl(ASTContext &context, const char *name,
+                   DXIL::SubobjectKind kind,
+                   DXIL::HitGroupType hgtype = (DXIL::HitGroupType)(-1)) {
   IdentifierInfo &id =
       context.Idents.get(StringRef(name), tok::TokenKind::identifier);
   CXXRecordDecl *decl = CXXRecordDecl::Create(
       context, TagTypeKind::TTK_Struct, context.getTranslationUnitDecl(), NoLoc,
       NoLoc, &id, nullptr, DelayTypeCreationTrue);
+  decl->addAttr(
+      HLSLSubObjectAttr::CreateImplicit(context, (unsigned)kind, (int)hgtype));
```

Comment:
Let's use `static_cast` for casts, according to the coding standards.

```suggestion
  decl->addAttr(HLSLSubObjectAttr::CreateImplicit(
      context, static_cast<unsigned>(Kind), static_cast<int>(HGT)));
```